File: sdks/kotlin/src/main/kotlin/com/puxbay/models/Customer.kt
package com.puxbay.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Customer(
    val id: String,
    val name: String,
    val email: String? = null,
    val phone: String? = null,
    val address: String? = null,
    @SerialName("customer_type") val customerType: String,
    @SerialName("loyalty_points") val loyaltyPoints: Int,
    @SerialName("store_credit_balance") val storeCreditBalance: Double,
    @SerialName("total_spend") val totalSpend: Double,
    val tier: String? = null,
    @SerialName("tier_name") val tierName: String? = null,
    @SerialName("marketing_opt_in") val marketingOptIn: Boolean = false,
    @SerialName("created_at") val createdAt: String
)

File: sdks/swift/Sources/PuxbaySDK/Models/Customer.swift
import Foundation

public struct Customer: Codable {
    public let id: String
    public let name: String
    public let email: String?
    public let phone: String?
    public let address: String?
    public let customerType: String
    public let loyaltyPoints: Int
    public let storeCreditBalance: Double
    public let totalSpend: Double
    public let tier: String?
    public let tierName: String?
    public let marketingOptIn: Bool?
    public let createdAt: String
}
